test/test_nn.py

```diff
@@ -443,6 +443,22 @@ def __init__(self):
         self.assertEqual(num_params(n), 3)
         self.assertEqual(num_params(s), 3)
 
+    def test_children(self):
```

Comment:
can you make the model deeper so that it checks that only direct children are returned?